From 029907a8e8fabc0c63fe6d007de1699d0b30304e Mon Sep 17 00:00:00 2001
From: Pierre Marzin <pierre.marzin@iot.bzh>
Date: Mon, 9 Mar 2020 17:35:44 +0100
Subject: [PATCH] config: rcar: add optimisation

 * Add boot timing info summary before booting OS
 * show print MMC env
 * Remove MMC debug info
 * Disable USB, EXT2, NAND, NET support
 * SILENT console
 * Disable unused compression mode
 * Disable unused commands
 * Optimize libc for speed
 * Enable FIT support
 * Disable default boot variable (not used if optimized on MMC)

Signed-off-by: Pierre Marzin <pierre.marzin@iot.bzh>
---
 configs/r8a7796_ulcb_defconfig     | 52 ++++++++++++++----------------
 include/configs/rcar-gen3-common.h |  5 ---
 include/configs/ulcb.h             |  4 +--
 3 files changed, 27 insertions(+), 34 deletions(-)

diff --git a/configs/r8a7796_ulcb_defconfig b/configs/r8a7796_ulcb_defconfig
index 5b8542ac41..1434cf0515 100644
--- a/configs/r8a7796_ulcb_defconfig
+++ b/configs/r8a7796_ulcb_defconfig
@@ -5,34 +5,38 @@ CONFIG_SYS_MALLOC_F_LEN=0x2000
 CONFIG_RCAR_GEN3=y
 CONFIG_R8A7796=y
 CONFIG_TARGET_ULCB=y
-CONFIG_SMBIOS_PRODUCT_NAME=""
+CONFIG_IDENT_STRING=" - BOOT OPTIMIZATION"
+# CONFIG_SYS_MALLOC_CLEAR_ON_INIT is not set
 CONFIG_FIT=y
 # CONFIG_ARCH_FIXUP_FDT_MEMORY is not set
 CONFIG_USE_BOOTARGS=y
-CONFIG_BOOTARGS="root=/dev/nfs rw nfsroot=192.168.0.1:/export/rfs ip=192.168.0.20"
+CONFIG_SILENT_CONSOLE=y
+CONFIG_SILENT_U_BOOT_ONLY=y
+CONFIG_SYS_CONSOLE_INFO_QUIET=y
 CONFIG_SUPPORT_RAW_INITRD=y
 CONFIG_DEFAULT_FDT_FILE="r8a7796-m3ulcb.dtb"
 CONFIG_VERSION_VARIABLE=y
 CONFIG_HUSH_PARSER=y
 CONFIG_CMD_BOOTZ=y
+# CONFIG_CMD_ELF is not set
+# CONFIG_CMD_GO is not set
+# CONFIG_CMD_XIMG is not set
 CONFIG_CMD_ASKENV=y
+# CONFIG_CMD_EXPORTENV is not set
+# CONFIG_CMD_IMPORTENV is not set
+# CONFIG_CMD_LZMADEC is not set
+# CONFIG_CMD_UNZIP is not set
 CONFIG_CMD_GPIO=y
-CONFIG_CMD_GPT=y
-CONFIG_CMD_I2C=y
 CONFIG_CMD_MMC=y
-CONFIG_CMD_USB=y
-CONFIG_CMD_DHCP=y
-CONFIG_CMD_MII=y
-CONFIG_CMD_PING=y
 CONFIG_CMD_CACHE=y
-CONFIG_CMD_EXT2=y
 CONFIG_CMD_EXT4=y
-CONFIG_CMD_EXT4_WRITE=y
-CONFIG_CMD_FAT=y
 CONFIG_CMD_FS_GENERIC=y
+CONFIG_DOS_PARTITION=y
+CONFIG_PARTITION_UUIDS=y
 CONFIG_OF_CONTROL=y
 CONFIG_DEFAULT_DEVICE_TREE="r8a7796-m3ulcb-u-boot"
 CONFIG_ENV_IS_IN_MMC=y
+# CONFIG_NET is not set
 CONFIG_REGMAP=y
 CONFIG_SYSCON=y
 CONFIG_CLK=y
@@ -42,19 +46,14 @@ CONFIG_RCAR_GPIO=y
 CONFIG_DM_I2C=y
 CONFIG_SYS_I2C_RCAR_IIC=y
 CONFIG_DM_MMC=y
+# CONFIG_MMC_QUIRKS is not set
+# CONFIG_MMC_HW_PARTITIONING is not set
 CONFIG_MMC_IO_VOLTAGE=y
 CONFIG_MMC_UHS_SUPPORT=y
 CONFIG_MMC_HS200_SUPPORT=y
+# CONFIG_MMC_VERBOSE is not set
 CONFIG_RENESAS_SDHI=y
-CONFIG_MTD=y
-CONFIG_MTD_NOR_FLASH=y
-CONFIG_MTD_DEVICE=y
-CONFIG_CFI_FLASH=y
-CONFIG_RENESAS_RPC_HF=y
-CONFIG_PHY_MICREL=y
-CONFIG_PHY_MICREL_KSZ90X1=y
-CONFIG_DM_ETH=y
-CONFIG_RENESAS_RAVB=y
+# CONFIG_PHY_RCAR_GEN3 is not set
 CONFIG_PINCTRL=y
 CONFIG_PINCONF=y
 CONFIG_PINCTRL_PFC=y
@@ -63,11 +62,10 @@ CONFIG_DM_REGULATOR_FIXED=y
 CONFIG_DM_REGULATOR_GPIO=y
 CONFIG_SCIF_CONSOLE=y
 CONFIG_SYSRESET=y
-CONFIG_USB=y
-CONFIG_DM_USB=y
-CONFIG_USB_XHCI_HCD=y
-CONFIG_USB_EHCI_HCD=y
-CONFIG_USB_EHCI_GENERIC=y
-CONFIG_USB_STORAGE=y
+CONFIG_EXT4_WRITE=y
+CONFIG_CC_OPTIMIZE_LIBS_FOR_SPEED=y
+CONFIG_REGEX=y
+CONFIG_LZ4=y
+CONFIG_LZO=y
 CONFIG_OF_LIBFDT_OVERLAY=y
-CONFIG_SMBIOS_MANUFACTURER=""
+# CONFIG_EFI_LOADER is not set
diff --git a/include/configs/rcar-gen3-common.h b/include/configs/rcar-gen3-common.h
index e8999ffb0c..287b98fc05 100644
--- a/include/configs/rcar-gen3-common.h
+++ b/include/configs/rcar-gen3-common.h
@@ -63,9 +63,4 @@
 	"fdt_high=0xffffffffffffffff\0"	\
 	"initrd_high=0xffffffffffffffff\0"
 
-#define CONFIG_BOOTCOMMAND	\
-	"tftp 0x48080000 Image; " \
-	"tftp 0x48000000 Image-"CONFIG_DEFAULT_FDT_FILE"; " \
-	"booti 0x48080000 - 0x48000000"
-
 #endif	/* __RCAR_GEN3_COMMON_H */
diff --git a/include/configs/ulcb.h b/include/configs/ulcb.h
index f12055ebad..eb68d425f0 100644
--- a/include/configs/ulcb.h
+++ b/include/configs/ulcb.h
@@ -12,8 +12,8 @@
 #include "rcar-gen3-common.h"
 
 /* Ethernet RAVB */
-#define CONFIG_BITBANGMII
-#define CONFIG_BITBANGMII_MULTI
+/*#define CONFIG_BITBANGMII*/
+/*#define CONFIG_BITBANGMII_MULTI*/
 
 /* Generic Timer Definitions (use in assembler source) */
 #define COUNTER_FREQUENCY	0xFE502A	/* 16.66MHz from CPclk */
-- 
2.17.1

